framework/core/src/Extend/Theme.php

```diff
@@ -100,12 +101,53 @@ public function addCustomLessFunction(string $functionName, callable $callable):
         return $this;
     }
 
+    /**
+     * Defines a new Less variable to be accessible in all Less files.
+     *
+     * This can be useful for styling based on a backend setting, for example.
+     *
+     * Remember that Less is not recompiled when settings values change, so you
+     * may need to hook into a Saved event to recompile stylesheets when the
+     * value of one of your extension's settings changes in the database.
+     *
+     * Please note the value returned from the callable will be inserted directly
+     * into the Less source. If it is unsafe in some way (e.g., contains a
+     * semi-colon), this will result in potential security issues with your
+     * stylesheet.
+     *
+     * Likewise, if you need your variable to be a string, you should surround it
+     * with quotes yourself.
+     *
+     * ```php
+     * (new Extend\Theme())
+     *   ->addCustomLessVariable('my-extension__show-thing', function () {
+     *     $settings = resolve(SettingsRepositoryInterface::class);
+     *     return boolval($settings->get('my-extension.show_thing', false));
+     *   })
+     * ```
+     *
+     * @param string $variableName Name of the variable identifier.
+     * @param callable $value The PHP function to run, which returns the value for the variable.
+     *
+     * @return self
+     */
```

Comment:
can we explicitly mention to use the `registerLessConfigVar` in the case of exposing a setting? probably also use a different example to avoid confusion